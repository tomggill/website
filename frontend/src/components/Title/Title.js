import React from 'react';
import "../../styles/styles.css"

const Title = ({title, style}) => {
  return (
    <h1 className="customTitle" style ={style}>
      {title}
    </h1>
  );
}

export default Title;