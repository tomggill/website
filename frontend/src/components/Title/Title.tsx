import React from 'react';
import '../../styles/styles.css';

interface TitleProps {
  title: string;
}

function Title(props: TitleProps) {
  return (
    <h1 className="customTitle">
      {props.title}
    </h1>
  );
}

export default Title;
