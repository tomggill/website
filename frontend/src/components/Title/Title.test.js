import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Title from './Title';

test('Renders title component', () => {
  const titleText = 'TestText';
  render(<Title title={titleText} />);
  const linkElement = screen.getByText(titleText);
  expect(linkElement).toBeInTheDocument();
});

test('Title snapshot test', () => {
  const component = renderer.create(<Title text="TestText" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
