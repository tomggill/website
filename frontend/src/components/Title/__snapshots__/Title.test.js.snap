// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title snapshot test 1`] = `
<h1
  className="customTitle"
/>
`;
